webapp/src/components/ImageFieldEditor.tsx

```diff
@@ -0,0 +1,326 @@
+import * as React from "react";
+
+import { FieldEditorComponent } from '../blocklyFieldView';
+import { ImageEditor } from "./ImageEditor/ImageEditor";
+import { Bitmap, imageLiteralToBitmap } from './ImageEditor/store/bitmap';
+import { setTelemetryFunction } from './ImageEditor/store/imageReducer';
+
+export interface ImageFieldEditorState {
+    galleryVisible: boolean;
+    galleryFilter?: string;
+}
+
+interface GalleryItem {
+    qName: string;
+    src: string;
+    alt: string;
+    tags: string[];
+}
+
+export class ImageFieldEditor extends React.Component<{}, ImageFieldEditorState> implements FieldEditorComponent {
+    protected blocksInfo: pxtc.BlocksInfo;
+    protected ref: ImageEditor;
+    protected closeEditor: () => void;
+
+    constructor(props: {}) {
+        super(props);
+
+        this.state = {
+            galleryVisible: false
+        };
+        setTelemetryFunction(tickImageEditorEvent);
+    }
+
+    render() {
+        return <div className="image-editor-wrapper">
+            <div className="gallery-editor-header">
+                <div className={`gallery-editor-toggle ${this.state.galleryVisible ? "right" : "left"} ${pxt.BrowserUtils.isEdge() ? "edge" : ""}`} onClick={this.toggleGallery} role="button" aria-pressed={this.state.galleryVisible}>
```

Comment:
could consider moving this up the DOM--a general "edge" css class might be useful in other scenarios too